tsl/test/expected/hypercore_trigger.out

```diff
@@ -0,0 +1,274 @@
+ _timescaledb_internal._hyper_1_4_chunk
+ _timescaledb_internal._hyper_1_5_chunk
+ _timescaledb_internal._hyper_1_6_chunk
+(6 rows)
+
+select chunk_name, compression_status from chunk_compression_stats(:'hypertable');
+    chunk_name    | compression_status 
+------------------+--------------------
+ _hyper_1_1_chunk | Compressed
+ _hyper_1_2_chunk | Compressed
+ _hyper_1_3_chunk | Compressed
+ _hyper_1_4_chunk | Compressed
+ _hyper_1_5_chunk | Compressed
+ _hyper_1_6_chunk | Compressed
+(6 rows)
+
+with the_info as (
+     select min(created_at) min_created_at,
+     	    max(created_at) max_created_at
+       from :hypertable
+)
+select min_created_at,
+       max_created_at,
+       '1m'::interval + min_created_at + (max_created_at - min_created_at) as mid_created_at,
+       '1m'::interval + max_created_at + (max_created_at - min_created_at) as post_created_at
+  from the_info \gset
+-- Insert a bunch of rows to make sure that we have a mix of
+-- uncompressed, partially compressed, and fully compressed
+-- chunks. Note that there is is an overlap between the compressed and
+-- uncompressed start and end timestamps.
+insert into :hypertable (created_at, location_id, device_id, owner_id, temp, humidity)
+select t, ceil(random()*10), ceil(random()*30), ceil(random() * 5), random()*40, random()*100
+from generate_series(:'mid_created_at'::timestamptz, :'post_created_at', '15m') t;
+-- Start by testing insert triggers for both statements and rows. In
+-- this case, the trigger will just save away the rows into a separate
+-- table and could that we get the same number of rows with the same
+-- values.
+create trigger save_insert_row_trg before insert on :chunk1 for each row execute function save_row();
+create trigger count_inserts_trg before insert on :chunk1 for each statement execute function count_ops();
+insert into :chunk1(created_at, location_id, device_id, owner_id, temp, humidity)
+  values
+	('2022-06-01 00:01:23', 999, 666, 111, 3.14, 3.14),
+	('2022-06-01 00:02:23', 999, 666, 111, 3.14, 3.14);
+select * from saved_rows where kind = 'INSERT';
+ metric_id |          created_at          | location_id | owner_id | device_id | temp | humidity | new_row |  kind  
+-----------+------------------------------+-------------+----------+-----------+------+----------+---------+--------
+     11523 | Wed Jun 01 00:01:23 2022 PDT |         999 |      111 |       666 | 3.14 |     3.14 | t       | INSERT
+     11524 | Wed Jun 01 00:02:23 2022 PDT |         999 |      111 |       666 | 3.14 |     3.14 | t       | INSERT
+(2 rows)
+
+select sum(inserts), sum(updates), sum(deletes) from count_stmt;
+ sum | sum | sum 
+-----+-----+-----
+   1 |   0 |   0
+(1 row)
+
+-- Run update tests
+create trigger save_update_row_trg before update on :chunk1 for each row execute function save_row();
+create trigger count_update_trg before update on :chunk1 for each statement execute function count_ops();
+update :chunk1 set temp = 9.99 where device_id = 666;
```

Comment:
Should we test upserts as well? Do you think it would make a difference or have the same semantics as update?

Reply:
Hmmm... and what about MERGE statement?

Reply:
Seems reasonable to add.

Reply:
Done.

Reply:
> Should we test upserts as well? Do you think it would make a difference or have the same semantics as update?

Added.

Reply:
> Hmmm... and what about MERGE statement?

Hmm... if a MERGE statement did not add do that, would that be a fault in our code or in PostgreSQL code? I realize the same actually applies to upserts.

Reply:
> > Hmmm... and what about MERGE statement?
> 
> Hmm... if a MERGE statement did not add do that, would that be a fault in our code or in PostgreSQL code? I realize the same actually applies to upserts.

Turns out MERGE is not supported for UPDATE and DELETE and generates an error. We have tests for that in `hypercore_merge.sql`. I added a test for INSERT.